plugins/right-sizer/src/controller/controller.go

```diff
@@ -284,34 +297,38 @@ func (c *Controller) processOOMKilledPod(pod *core.Pod) {
 		reportItem := c.reportBuilder.CreateRightSizerReportItem(podControllerObject, containerInfo)
 
 		if c.config.updateMemoryLimits {
-			if c.config.updateMemoryLimitsMinimumOOMs > 0 && reportItem.NumOOMs >= c.config.updateMemoryLimitsMinimumOOMs {
-				containerMemoryLimits := containerInfo.Resources.Limits.Memory()
-				newContainerMemoryLimits := reportItem.IncrementMemory(containerMemoryLimits, c.config.updateMemoryLimitsIncrement, c.config.updateMemoryLimitsMax)
-				if !newContainerMemoryLimits.IsZero() {
-					glog.V(1).Infof("%s has memory  limit %v, updating to %v", reportItem, containerMemoryLimits, newContainerMemoryLimits)
-					patchedResource, err := util.PatchContainerMemoryLimits(c.kubeClientResources, podControllerObject, reportItem.ResourceContainer, newContainerMemoryLimits)
-					if err != nil {
-						// EndingMemory remains the previously set StartingMemory value.
-						glog.Errorf("error patching %s memory limits: %v", reportItem, err)
-					} else {
-						// The post-patch memory limits may be different than the limits in our patch.
-						// I.E. We patch with 15655155794400u, post-patch shows 15655155795m
-						// Update  report item with the actual; post-patch memory.
-						patchedContainer, _, _, foundPatchedContainer, err := util.FindContainerInUnstructured(patchedResource, reportItem.ResourceContainer)
-						if !foundPatchedContainer || err != nil {
-							glog.Errorf("unable to find the container %s in patched report item %s, and will have to set report memory limits to the value sent in the patch: %s - potential error = %v", reportItem.ResourceContainer, reportItem, newContainerMemoryLimits, err)
-							reportItem.EndingMemory = newContainerMemoryLimits
+			if len(c.config.allowedUpdateNamespaces) > 0 && funk.ContainsString(c.config.allowedUpdateNamespaces, pod.Namespace) {
```

Comment:
I'd suggest trying to break up this function a bit - it's pretty long and deeply nested.